src/Compilers/CSharp/Test/Emit/PDB/PortablePdbTests.cs

```diff
@@ -209,6 +210,151 @@ public static void Main()
             }
         }
 
+        [Fact]
+        public void PortablePdb_DeterministicCompilation()
+        {
+            var referenceTimestamp = DateTime.Now;
+
+            var referenceCompilation = CreateCompilation(
+@"public struct StructWithReference
+{
+    string PrivateData;
+}
+public struct StructWithValue
+{
+    int PrivateData;
+}", options: TestOptions.DebugDll.WithCurrentLocalTime(referenceTimestamp));
```

Comment:
>WithCurrentLocalTime(referenceTimestamp) [](start = 34, length = 40)

WithCurrentLocalTime shouldn't be needed. It's only used for time-based assembly version generation. 

The timestamp in the PE headers is not this timestamp. For deterministic build it's actually not timestamp at all. It's just 4B of SHA2 hash of the content of the binary.

Reply:
Thanks for the clarification. I changed to emit a PE into memory and pull information from there.